### PR TITLE
refactor(step-tracker): refactor step tracker

### PR DESCRIPTION
**Pull request by kherP**
- revamped step tracker to support accessibility according to this [guide ](https://a11y-guidelines.orange.com/en/web/components-examples/stepbar/)
- refactored css styles with BEM methodology
- added improvements:
  - vocalization support for accessibility
  - checkmark for last step if step is more than index of last step
- added accessibility example in documentation

* side note: there will be no keyboard interaction in this component but the click function of this component is retained to prevent breaking changes

re #628

### CODE REVIEW

File: docs/src/pages/docs/steptracker.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React from "react";
 import Docs from "@common/Docs";
 import { DynamicFormOption, DynamicFormSection, useDynamicForm } from "@sebgroup/react-components/hooks/useDynamicForm";
 import { StepTracker } from "@sebgroup/react-components/StepTracker";
-import StepLabel, { StepLabelProps } from "@sebgroup/react-components/StepTracker/StepLabel";
+import { StepLabel, StepLabelProps } from "@sebgroup/react-components/StepTracker";
```
Comment (eweseong) [inline docs/src/pages/docs/steptracker.tsx:5]: let squash the import into line `4`

File: lib/src/StepTracker/steplabel.scss [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+@mixin tracker-line() {
+    &:not(:last-of-type) {
+        &::after {
+            @content;
+        }
+    }
+}
+
+.#{$step-label-component-name} {
```
Comment (eweseong) [inline lib/src/StepTracker/steplabel.scss:20]: maybe we can utilise the ampersand trick to control the nesting of BEM naming convention
https://css-tricks.com/using-sass-control-scope-bem-naming/#so-how-does-this-relate-to-sass

```scss
.step-label {
    $step: &;

    display: flex;
    flex: 1;
    position: relative;
    gap: 0.5rem;
    align-items: center;

    @include tracker-line() {
        content: "";
        position: absolute;
        height: $step-line-height;
        top: calc(50% + #{math.div($step-number-height, 2)});
        left: calc(#{math.div($step-number-width, 2)} - #{math.div($step-line, 2)});
        width: $step-line;
        background: linear-gradient(to left, $gray-300 50%, $blue-darker 50%) right;
        background-size: 300% 100%;
        background-position: right;
        transition: background-position 200ms ease-out;
    }

    &:hover {
        cursor: pointer;
    }

    &__label {
        flex: 0 0 $step-number-width;
        border-radius: 50%;
        border: 2px solid $gray-300;
        background-color: $gray-300;
        height: $step-number-height;
        width: $step-number-width;
        text-align: center;

        &--numbered {
            line-height: unset;
            font-size: $font-size-xs;
        }
    }

    &__name {
        position: relative;
        font-size: $font-size-sm;
        flex: 1;
        line-height: $step-number-height;
    }

    &--active {
        #{step}__label {
            border-color: $blue-darker;
        }

        #{$step}__name {
            color: $blue-darker;
        }
    }

    &--completed {
        @include tracker-line() {
            background-position: left;
        }

        #{$step}__label {
            color: $white;
            border-color: $blue-darker;
            background-color: $blue-darker;

            > svg {
                width: $font-size-xs;
                height: $font-size-xs;
                vertical-align: $step-line;
            }
        }
    }

    &--left {
        flex-direction: row-reverse;

        @include tracker-line() {
            left: unset;
            right: calc(#{math.div($step-number-width, 2)} - #{math.div($step-line, 2)});
        }
    }

    &--bottom,
    &--top {
        flex-direction: column;
        padding-bottom: 0;

        @include tracker-line() {
            height: $step-line;
            top: calc(#{math.div($step-number-width, 2)} - #{math.div($step-line, 2)});
            left: calc(50% + #{math.div($step-number-width, 2)});
            width: $step-line-height;
        }
    }

    &--bottom {
        flex-direction: column;
    }

    &--top {
        flex-direction: column-reverse;

        @include tracker-line() {
            top: unset;
            bottom: calc(#{math.div($step-number-width, 2)});
            left: calc(50% + #{math.div($step-number-width, 2)});
            width: $step-line-height;
        }
    }
}
```

Comment (kherP) [reply lib/src/StepTracker/steplabel.scss:20]: Nice! Will do that

File: lib/src/StepTracker/steplabel.scss [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
 $step-number-width: $step-number-height;
 $step-line: 4px;
 $step-line-height: 100%;
 
+@mixin component-root() {
+    .step-label {
+        @content;
+    }
+}
+
```
Comment (eweseong) [inline lib/src/StepTracker/steplabel.scss:14]: nice idea! will steal this for future reference :D

**Review by eweseong — APPROVED** [review]

**Pull request merged**